eva-server/src/main/java/uk/ac/ebi/variation/eva/server/ws/EvaWSServer.java

```diff
@@ -45,118 +40,77 @@
-        this.uriInfo = uriInfo;
-        this.httpServletRequest = hsr;
-
-        this.startTime = System.currentTimeMillis();
-        this.queryResponse = new QueryResponse();
+    protected void initializeQueryOptions() {
         this.queryOptions = new QueryOptions();
+        Map<String, String[]> multivaluedMap = httpServletRequest.getParameterMap();
+        
+        boolean metadata = (multivaluedMap.get("metadata") != null) ? multivaluedMap.get("metadata")[0].equals("true") : true ;
+        int limit = (multivaluedMap.get("limit") != null) ? Integer.parseInt(multivaluedMap.get("limit")[0]) : -1;
+        int skip = (multivaluedMap.get("skip") != null) ? Integer.parseInt(multivaluedMap.get("skip")[0]) : -1;
+        boolean count = (multivaluedMap.get("count") != null) ? multivaluedMap.get("count")[0].equals("true") : false ;
 
-        logger.info("EvaWSServer: in 'constructor'");
-    }
-
-    protected void checkParams() throws VersionException, SpeciesException {
-        // TODO A Version and Species checker must be implemented
-        if (version == null || !version.equals("v1")) {
-            throw new VersionException("Version not valid: '" + version + "'");
-        }
-        if (species == null || species.isEmpty()/*|| !isValidSpecies(species)*/) {
-            throw new SpeciesException("Species not valid: '" + species + "'");
-        }
-
-        MultivaluedMap<String, String> multivaluedMap = uriInfo.getQueryParameters();
-        queryOptions.put("species", species);
-
-        queryOptions.put("metadata", (multivaluedMap.get("metadata") != null) ? multivaluedMap.get("metadata").get(0).equals("true") : true);
-        queryOptions.put("exclude", (exclude != null && !exclude.equals("")) ? Splitter.on(",").splitToList(exclude) : null);
-        queryOptions.put("include", (include != null && !include.equals("")) ? Splitter.on(",").splitToList(include) : null);
+        String[] exclude = multivaluedMap.get("exclude");
+        String[] include = multivaluedMap.get("include");
+        
+        queryOptions.put("metadata", metadata);
+        queryOptions.put("exclude", (exclude != null && exclude.length > 0) ? Splitter.on(",").splitToList(exclude[0]) : null);
+        queryOptions.put("include", (include != null && include.length > 0) ? Splitter.on(",").splitToList(include[0]) : null);
         queryOptions.put("limit", (limit > 0) ? limit : -1);
         queryOptions.put("skip", (skip > 0) ? skip : -1);
-        queryOptions.put("count", (count != null && !count.equals("")) ? Boolean.parseBoolean(count) : false);
+        queryOptions.put("count", count);
+        System.out.println(queryOptions.toJson());
     }
 
-    protected Response createOkResponse(Object obj) {
-        endTime = System.currentTimeMillis() - startTime;
-        queryResponse.setTime(new Long(endTime - startTime).intValue());
+    protected QueryResponse setQueryResponse(Object obj) {
```

Comment:
I'm not sure if this is the best name since with setting methods you don't expect a return..